### PR TITLE
docs: add an overview of the creation and querying of snapshots

### PR DESCRIPTION
**Pull request by joshieDo**
Provides a current overview on how each component interacts with each other when creating a snapshot, or querying a bunch of them.

The glossary provides a brief summary of each component/struct alongside a link to the source code (which is documented), for a more detailed view.

### CODE REVIEW

File: crates/snapshot/README.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+```
+</details>
+
+
+### Glossary
+In descending order of abstraction hierarchy:
+
+[`Snapshotter`](../../crates/snapshot/src/snapshotter.rs#L20): A `reth` background service that **copies** data from the database to new snapshot files when the block height reaches a certain threshold (e.g., `500_000th`). Upon completion, it dispatches a notification about the higher snapshotted block to `HighestSnapshotTracker` channel. **It DOES NOT remove data from the database.**
+
+[`HighestSnapshotTracker`](../../crates/snapshot/src/snapshotter.rs#L22): A channel utilized by `Snapshotter` to announce the newest snapshot block to all components with a listener: `Pruner` (to know which additional tables can be pruned) and `DatabaseProvider`  (to know which data can be queried from the snapshots).
```
Comment (mattsse) [inline crates/snapshot/README.md:73]: I see, so the `HighestSnapshotTracker` channel acts both as a notification about changed snapshot tracker (for pruner) and to track the highest snapshot (for database)

File: crates/snapshot/README.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+<details>
+  <summary>Creation diagram (<code>Snapshotter</code>)</summary>
+
+```mermaid
+graph TD;
+    I("BLOCK_HEIGHT % 500_000 == 0")--triggers-->SP(Snapshotter)
+    SP --> |triggers| SH["create_snapshot(block_range, SnapshotSegment::Headers)"]
+    SP --> |triggers| ST["create_snapshot(block_range, SnapshotSegment::Transactions)"]
+    SP --> |triggers| SR["create_snapshot(block_range, SnapshotSegment::Receipts)"]
```
Comment (mattsse) [inline crates/snapshot/README.md:19]: this should emphasize that this is a list of "segments", perhaps with an additional `...` or a label

**Review by mattsse — CHANGES_REQUESTED** [review]
thanks for this,

after looking at the graphs I have a better understanding now.

**Review by mattsse — APPROVED** [review]
lgtm, pending more reviews

**Review by shekhirin — APPROVED** [review]
LGTM, only nits, really like these diagrams!

File: crates/snapshot/README.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+[`SnapshotSegment`](../../crates/primitives/src/snapshot/segment.rs#L10) Each snapshot file only contains data of a specific segment, e.g., `Headers`, `Transactions`, or `Receipts`.
+
+[`NippyJarCursor`](../../crates/storage/nippy-jar/src/cursor.rs#L12) Accessor of data in a `NippyJar` file. It enables queries either by row number (e.g., block number 1) or by a predefined key not part of the file (e.g., transaction hashes). If a file has multiple columns (e.g., `Tx | TxSender | Signature`), and one wishes to access only one of the column values, this can be accomplished by bitmasks. (e.g., for `TxSender`, the mask would be `0b010`).
+
+[`NippyJar`](../../crates/storage/nippy-jar/src/lib.rs#57) A create-only file format. No data can be appended after creation. It supports multiple columns, compression (e.g., Zstd (with and without dictionaries), lz4, uncompressed) and inclusion filters (e.g., cuckoo filter: `is hash X part of this dataset`). Snapshots are organized by block ranges. (e.g., `TransactionSnapshot_500_000.jar` contains a transaction per row for all transactions until block `500_000`). For more check the struct documentation.
```
[the remark this answers is not in the document]
Comment (shekhirin) [reply crates/snapshot/README.md:87]: oh btw does it actually contain `0..=500_000` or `0..500_000`? `0` is genesis, but we snapshot it too iiuc?

File: crates/snapshot/README.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  <summary>Query diagram (<code>Provider</code>)</summary>
+
+```mermaid
+graph TD;
+    RPC-->P
+    P("Provider::header(block_number)")-->PF(ProviderFactory)
+    PF--shares-->SP1("Arc(SnapshotProvider)")
+    SP1--shares-->PD(DatabaseProvider)
+    PF--creates-->PD
+    PD--check `HighestSnapshotTracker`-->PD
```
[the remark this answers is not in the document]
Comment (joshieDo) [reply crates/snapshot/README.md:53]: it becomes ugly in the diagram, i dont know how to force it to the side lol

**Pull request merged**